src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -711,6 +712,17 @@ public Project clone(UUID from, String newVersion, boolean includeTags, boolean
             }
         }
 
+     
+        final List<PolicyViolation> sourcePolicyViolations = getAllPolicyViolations(source);
```

Comment:
I think this only makes sense when the option to clone components is enabled, otherwise this won't work, or? Therefore it should be in the corresponding condition.
I think even further, there should be an option "Clone Policy Violations" similar to "clone audit history", and this should only be done when "clone components" + "Clone policy violations" is enabled.
The new option should then also be added to the clone UI.